File: app/src/main/java/com/omar/retromp3recorder/app/repo/BitRateRepo.java
package com.omar.retromp3recorder.app.repo;

import com.omar.retromp3recorder.app.recorder.RecorderDefaults;
import com.omar.retromp3recorder.app.recorder.VoiceRecorder;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Observable;
import io.reactivex.subjects.BehaviorSubject;

@Singleton
public class BitRateRepo {

    private final BehaviorSubject<VoiceRecorder.BitRate> stateSubject = BehaviorSubject.create();

    @Inject
    public BitRateRepo(RecorderDefaults defaults){
        stateSubject.onNext(defaults.bitRate);
    }

    public void newValue(VoiceRecorder.BitRate bitRate) {
        stateSubject.onNext(bitRate);
    }

    public Observable<VoiceRecorder.BitRate> observe() {
        return stateSubject;
    }
}

File: app/src/main/java/com/omar/retromp3recorder/app/state/BitRateRepo.kt
package com.omar.retromp3recorder.app.state

import com.omar.retromp3recorder.recorder.Mp3VoiceRecorder
import com.omar.retromp3recorder.recorder.RecorderDefaults
import io.reactivex.Observable
import io.reactivex.subjects.BehaviorSubject
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class BitRateRepo @Inject constructor(defaults: RecorderDefaults) {
    private val stateSubject = BehaviorSubject.create<Mp3VoiceRecorder.BitRate>()

    init {
        stateSubject.onNext(defaults.bitRate)
    }

    fun newValue(bitRate: Mp3VoiceRecorder.BitRate) {
        stateSubject.onNext(bitRate)
    }

    fun observe(): Observable<Mp3VoiceRecorder.BitRate> {
        return stateSubject
    }
}
